examples/bank/bank.go

```diff
@@ -0,0 +1,195 @@
+			if err != nil {
+				return err
+			}
+			toValue = readValue
+		}
+		// Update both accounts.
+		txn.Prepare(client.Put(proto.Key(from), []byte(fmt.Sprintf("%d", fromValue-amount))))
+		txn.Prepare(client.Put(proto.Key(to), []byte(fmt.Sprintf("%d", toValue+amount))))
+		return nil
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	atomic.AddInt32(&bank.numTransactions, 1)
+	return true
+}
+
+// Read the balances in all the accounts and return them.
+func (bank *Bank) readAllAccounts() []int64 {
+	balances := make([]int64, bank.numAccounts)
+	calls := make([]client.Call, bank.numAccounts)
+	txnOpts := &client.TransactionOptions{Name: "Reading all balances"}
+	err := bank.kvClient.RunTransaction(txnOpts, func(txn *client.Txn) error {
+		for i := 0; i < bank.numAccounts; i++ {
+			calls[i] = client.Get(proto.Key(makeAccountID(i)))
+		}
+		if err := txn.Run(calls...); err != nil {
+			log.Fatal(err)
+		}
+		// Copy responses into balances.
+		for i := 0; i < bank.numAccounts; i++ {
+			gr := calls[i].Reply.(*proto.GetResponse)
+			if gr.Value != nil && gr.Value.Bytes != nil {
+				balance, err := strconv.ParseInt(string(gr.Value.Bytes), 10, 64)
+				if err != nil {
+					log.Fatal(err)
+				}
+				balances[i] = balance
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	return balances
+}
+
+// continuouslyTransferMoney() keeps moving random amounts between
+// random accounts.
+func (bank *Bank) continuousMoneyTransfer() {
+	for {
+		from := makeAccountID(rand.Intn(bank.numAccounts))
+		to := makeAccountID(rand.Intn(bank.numAccounts))
+		exchange := rand.Int63n(100)
+		bank.moveMoney(from, to, exchange)
+	}
+}
+
+// Initialize all the bank accounts with cash.
+func (bank *Bank) initBankAccounts(cash int64) {
+	calls := make([]client.Call, bank.numAccounts)
+	for i := 0; i < bank.numAccounts; i++ {
+		calls[i] = client.Put(proto.Key(makeAccountID(i)), []byte(fmt.Sprintf("%d", cash)))
+	}
+	if err := bank.kvClient.Run(calls...); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func main() {
+	fmt.Printf("A simple program that keeps moving money between bank accounts\n\n")
+	// Run a test cockroach instance to represent the bank.
+	security.SetReadFileFn(securitytest.Asset)
+	serv := server.StartTestServer(nil)
+	defer serv.Stop()
+	// Initialize the bank.
+	var bank Bank
+	bank.numAccounts = 1000
+	// Key Value Client initialization.
+	sender, err := client.NewHTTPSender(serv.ServingAddr(), testutils.NewTestBaseContext())
+	if err != nil {
+		log.Fatal(err)
+	}
+	bank.kvClient = client.NewKV(nil, sender)
+	bank.kvClient.User = storage.UserRoot
+	// Initialize all the bank accounts.
+	const initCash = 1000
+	bank.initBankAccounts(initCash)
+
+	// Start all the money transfer routines.
+	const numTransferRoutines = 10
+	for i := 0; i < numTransferRoutines; i++ {
+		go bank.continuousMoneyTransfer()
+	}
+	// Sleep for a bit to allow money transfers to happen in the background and then exit.
+	time.Sleep(10 * time.Second)
```

Comment:
Rather than checking the balances once, I think I'd put this in a loop where you sleep for a second then wake up and check the balances. That way the demo just runs until the user ctrl-c's it.